libnetwork/drivers/overlay/ovmanager/ovmanager.go

```diff
@@ -159,42 +162,15 @@ func (d *driver) NetworkFree(id string) error {
 	return nil
 }
 
-func (n *network) obtainVxlanID(s *subnet) error {
-	var (
-		err error
-		vni uint64
-	)
-
-	n.Lock()
-	vni = uint64(s.vni)
-	n.Unlock()
-
-	if vni == 0 {
-		vni, err = n.driver.vxlanIdm.GetIDInRange(vxlanIDStart, vxlanIDEnd, true)
-		if err != nil {
-			return err
-		}
-
-		n.Lock()
-		s.vni = uint32(vni)
-		n.Unlock()
-		return nil
-	}
-
-	return n.driver.vxlanIdm.GetSpecificID(vni)
-}
-
 func (n *network) releaseVxlanID() {
-	n.Lock()
 	vnis := make([]uint32, 0, len(n.subnets))
```

Comment:
Given that we are holding the lock during the entirety of this function call now, does it make sense to create this intermediate slice?
Seems like this could be pretty expensive.

Reply:
Or is `Release` expensive enough that we shouldn't be holding the lock during this?

Reply:
> Given that we are holding the lock during the entirety of this function call now, does it make sense to create this intermediate slice?

No, it does not. Good catch, fixed!

`len(n.subnets)` should be small since the user has to enumerate all the subnets when creating the network, and `Release`/`Unset` shouldn't be too expensive. The bitmap is not safe for concurrent use so has to be accessed within a critical section. And if the bitmap operation does prove to be expensive enough to be a problem, I would solve that by replacing our bespoke bitmap implementation with [roaring](https://pkg.go.dev/github.com/RoaringBitmap/roaring).